server/src/main/java/org/apache/lucene/index/BaseSingleStarTreeBuilder.java

```diff
@@ -0,0 +1,660 @@
+            // For leaf node
+
+            if (node.startDocId == node.endDocId - 1) {
+                // If it has only one document, use it as the aggregated document
+                aggregatedStarTreeDocument = getStarTreeDocument(node.startDocId);
+                node.aggregatedDocId = node.startDocId;
+            } else {
+                // If it has multiple documents, aggregate all of them
+                for (int i = node.startDocId; i < node.endDocId; i++) {
+                    aggregatedStarTreeDocument = aggregateStarTreeDocument(aggregatedStarTreeDocument, getStarTreeDocument(i));
+                }
+                assert aggregatedStarTreeDocument != null;
+                for (int i = node.dimensionId + 1; i < numDimensions; i++) {
+                    aggregatedStarTreeDocument.dimensions[i] = STAR_IN_DOC_VALUES_INDEX;
+                }
+                node.aggregatedDocId = numDocs;
+                appendToStarTree(aggregatedStarTreeDocument);
+            }
+        } else {
+            // For non-leaf node
+            if (node.children.containsKey((long) StarTreeBuilderUtils.ALL)) {
+                // If it has star child, use the star child aggregated document directly
+                for (StarTreeBuilderUtils.TreeNode child : node.children.values()) {
+                    if (child.dimensionValue == StarTreeBuilderUtils.ALL) {
+                        aggregatedStarTreeDocument = createAggregatedDocs(child);
+                        node.aggregatedDocId = child.aggregatedDocId;
+                    } else {
+                        createAggregatedDocs(child);
+                    }
+                }
+            } else {
+                // If no star child exists, aggregate all aggregated documents from non-star children
+                for (StarTreeBuilderUtils.TreeNode child : node.children.values()) {
+                    aggregatedStarTreeDocument = aggregateStarTreeDocument(aggregatedStarTreeDocument, createAggregatedDocs(child));
+                }
+                assert aggregatedStarTreeDocument != null;
+                for (int i = node.dimensionId + 1; i < numDimensions; i++) {
+                    aggregatedStarTreeDocument.dimensions[i] = STAR_IN_DOC_VALUES_INDEX;
+                }
+                node.aggregatedDocId = numDocs;
+                appendToStarTree(aggregatedStarTreeDocument);
+            }
+        }
+        return aggregatedStarTreeDocument;
+    }
+
+    /**
+     * Handles the dimension of date time field type
+     *
+     * @param fieldName name of the field
+     * @param val       value of the field
+     * @return returns the converted dimension of the field to a particular granularity
+     */
+    private long handleDateDimension(final String fieldName, final long val) {
+        // TODO: handle timestamp granularity
+        return val;
+    }
+
+    public void close() throws IOException {
+        // boolean success = false;
```

Comment:
lets remove all commented code related to file formats as there is a possibility that it can completely change from POC.